xarray/tests/test_concat.py

```diff
@@ -18,61 +18,62 @@
 
 
 class TestConcatDataset:
-    def test_concat(self):
-        # TODO: simplify and split this test case
-
-        # drop the third dimension to keep things relatively understandable
-        data = create_test_data()
-        for k in list(data.variables):
-            if "dim3" in data[k].dims:
-                del data[k]
-
-        split_data = [data.isel(dim1=slice(3)), data.isel(dim1=slice(3, None))]
-        assert_identical(data, concat(split_data, "dim1"))
-
-        def rectify_dim_order(dataset):
-            # return a new dataset with all variable dimensions transposed into
-            # the order in which they are found in `data`
-            return Dataset(
-                {k: v.transpose(*data[k].dims) for k, v in dataset.data_vars.items()},
-                dataset.coords,
-                attrs=dataset.attrs,
-            )
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self.data = create_test_data().drop_dims("dim3")
```

Comment:
Obv no need to change but a hopefully helpful comment - this could very easily be a fixture called `data`, and then no need for using instance attributes

Reply:
How would I do that?

Reply:
```python

    @pytest.fixture
    def data(self):
        return create_test_data().drop_dims("dim3")
```

and then in the methods, e.g.

```python
    def test_concat(self, data):  # <- added 

        # data = self.data  <- no longer needed
        split_data = [data.isel(dim1=slice(3)), data.isel(dim1=slice(3, None))]
        assert_identical(data, concat(split_data, "dim1"))
```

Reply:
OK. done.